File: 1929/1929.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   1929.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/09/30 00:08:42 by jihoolee          #+#    #+#             */
/*   Updated: 2021/09/30 01:03:57 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>

using namespace std;

vector<bool>	eratos(int start, int end)
{
	vector<bool>	result(end + 1, 1);

	result[0] = false;
	result[1] = false;
	for (int i = 2; i * i <= end; i++)
	{
		if (result[i] == true)
			for (int j = 2; j * i <= end; j++)
				result[i * j] = false;
	}
	return (result);
}

int	main(void)
{
	int				start, end;
	vector<bool>	primes;

	cin >> start >> end;
	primes = eratos(start, end);
	for (int i = start; i <= end; i++)
		if (primes[i] == true)
			cout << i << '\n';
	return (0);
}

File: 1929/1929.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    1929.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2021/09/30 01:10:58 by jihoolee          #+#    #+#              #
#    Updated: 2021/09/30 01:36:25 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

start, end = map(int, input().split())

def eratos(start, end):
	end += 1
	prime = [True for _ in range (end)]
	prime[1] = False
	range_end = int(end ** 0.5) + 1
	for i in range(2, range_end + 1):
		if (prime[i]):
			for j in range(2 * i, end, i):
				prime[j] = False
	return (prime)

primes_lst = eratos(start, end)
for i in range(start, end + 1):
	if (primes_lst[i] == True):
		print(i)
